dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -433,6 +445,48 @@ private String channelName() {
         }
     }
 
+    private static final class ConsumptionTrackingRequestBody implements RequestBody {
+        private final RequestBody delegate;
+        private volatile boolean consumed;
+
+        ConsumptionTrackingRequestBody(RequestBody delegate) {
+            this.delegate = delegate;
+            this.consumed = false;
```

Comment:
booleans default to `false`, explicitly setting a volatile variable to its default variable is more expensive than the implicit assignment because it forces volatile assignment semantics unnecessarily (edge cases can apply if a constructor passes the partially-initialized `this` reference elsewhere, but that's not considered good practice anyhow).

We could write an error-prone rule for this, however there's a small legibility cost, and it doesn't matter much outside of hot performance-sensitive codepaths.

Reply:
Interesting consideration.

> edge cases can apply if a constructor passes the partially-initialized this reference elsewhere

I suppose this case is when, in the constructor, initialization is done in multiple threads which read the volatile variable. Seems scary to debug.